src/Metrics.h

```diff
@@ -457,17 +471,38 @@ class TopN final : public Metric
      * @param j json object
      * @param formatter std::function which takes a T as input (the type store it in top table) it needs to return a std::string
      */
-    void to_json(json &j, std::function<std::string(const T &)> formatter) const
+    void to_json(json &j, std::function<std::string(const T &)> formatter, Aggregate op = Aggregate::DEFAULT) const
     {
         auto section = json::array();
         auto items = _fi.get_frequent_items(datasketches::frequent_items_error_type::NO_FALSE_NEGATIVES);
         auto threshold = _get_threshold(items);
-        for (uint64_t i = 0; i < std::min(_top_count, items.size()); i++) {
-            if (items[i].get_estimate() >= threshold) {
-                section[i]["name"] = formatter(items[i].get_item());
-                section[i]["estimate"] = items[i].get_estimate();
-            } else {
-                break;
+        if (op == Aggregate::DEFAULT) {
```

Comment:
can we break this logic out so that we are not copy and pasting it between json and prometheus?

Reply:
done